addons/new-osio-addon/src/main/java/io/fabric8/launcher/osio/OsioMissionControl.java

```diff
@@ -65,39 +63,60 @@ public Boom launch(Projectile genericProjectile) throws IllegalArgumentException
         OsioProjectile projectile = (OsioProjectile) genericProjectile;
         GitRepository repository = gitSteps.createRepository(projectile);
 
+        executeCommonSteps(projectile, repository);
+
+        gitSteps.pushToGitRepository(projectile, repository);
+
+        openShiftSteps.triggerBuild(projectile);
+        return ImmutableBoom.builder()
+                .createdRepository(repository)
+                .createdProject(new OsioOpenShiftProject(projectile))
+                .build();
+    }
+
+    private void executeCommonSteps(OsioProjectile projectile, GitRepository repository) {
         openShiftSteps.createBuildConfig(projectile, repository);
         openShiftSteps.createJenkinsConfigMap(repository);
 
         // create webhook first so that push will trigger build
         gitSteps.createWebHooks(projectile, repository);
         gitSteps.pushToGitRepository(projectile, repository);
+    }
 
-        gitSteps.pushToGitRepository(projectile, repository);
+    public Boom launch(OsioProjectile projectile) {
+        GitRepository repository = gitSteps.findRepository(projectile);
+
+        executeCommonSteps(projectile, repository);
 
         openShiftSteps.triggerBuild(projectile);
+
         return ImmutableBoom.builder()
                 .createdRepository(repository)
-                .createdProject(new OpenShiftProject() {
-                    @Override
-                    public String getName() {
-                        return projectile.getOpenShiftProjectName();
-                    }
-
-                    @Override
-                    public URL getConsoleOverviewUrl() {
-                        return null;
-                    }
-
-                    @Override
-                    public List<OpenShiftResource> getResources() {
-                        return null;
-                    }
-                })
+                .createdProject(new OsioOpenShiftProject(projectile))
                 .build();
     }
 
+    private static class OsioOpenShiftProject implements OpenShiftProject {
```

Comment:
How about adding @Immutable to OpenShiftProject and use the generated ImmutableOpenShiftProject class instead?